### PR TITLE
[Merged by Bors] - Add topic level compression

### LINKED ISSUE #2249: Topic level compression
Right now, compression is only configured in the producer. We may want to have the ability to define the default topic level compression, and then producer will use that by default

### PR DESCRIPTION
**Pull request by morenol**
Closes #2249

### CODE REVIEW

File: crates/fluvio/src/producer/mod.rs [hunk 1]
```diff
@@ -288,7 +290,40 @@ impl TopicProducer {
...
             .spec;
         let partition_count = topic_spec.partitions();
-        let compression = config.compression;
+
+        let compression =
+            match topic_spec.get_compression_type() {
+                None | Some(CompressionType::Any) => config.compression.unwrap_or_default(),
+                Some(CompressionType::Gzip) => match config.compression {
+                    Some(Compression::Gzip) | None => Compression::Gzip,
+                    _ => return Err(FluvioError::Producer(ProducerError::InvalidConfiguration(
```
Comment (sehz) [inline crates/fluvio/src/producer/mod.rs:299]: Error message should include invalid compression for better debug ability

**Review by sehz — CHANGES_REQUESTED** [review]
Looks good overall.   Just comment about compression configuration.

File: crates/fluvio-controlplane-metadata/src/topic/spec.rs [hunk 2]
```diff
@@ -181,6 +189,8 @@ pub(crate) struct TopicSpecInner {
     cleanup_policy: Option<CleanupPolicy>,
     #[fluvio(min_version = 4)]
     storage: Option<TopicStorageConfig>,
+    #[fluvio(min_version = 6)]
+    compression_type: Option<CompressionType>,
```
[the remark this answers is not in the document]
Comment (morenol) [reply crates/fluvio-controlplane-metadata/src/topic/spec.rs:193]: but none is not the default, it is Any

[the remark this answers is not in the document]
Comment (sehz) [reply crates/fluvio-controlplane-metadata/src/topic/spec.rs:193]: So it shall be

**Review by sehz — APPROVED** [review]
LGTM!

**Pull request closed without merging**